daemon/qrexec-daemon.c

```diff
@@ -853,37 +1058,105 @@ static void handle_execute_service(
             policy_pending[policy_pending_slot].response_sent = RESPONSE_PENDING;
             return;
     }
-
-    for (i = 3; i < MAX_FDS; i++)
-        close(i);
-
-    result = connect_daemon_socket(remote_domain_id, remote_domain_name,
-                                   target_domain, service_name, request_id);
-    if (result >= 0) {
-        _exit(result);
+    if (close_range(3, ~0U, 0))
+        abort(); /* cannot happen */
+
+    char *user, *target, *requested_target;
+    int autostart;
+    connect_daemon_socket(remote_domain_name, target_domain, service_name,
+                          &user, &target, &requested_target, &autostart);
+    /* Replace the target domain with the version normalized by the policy engine */
+    target_domain = requested_target;
+    char *cmd = NULL;
+    bool disposable = false;
+    size_t resp_len;
+    /*
+     * If there was no service argument, pretend that an empty argument was
+     * provided by appending "+" to the service name.
+     */
+    const char *const trailer = strchr(service_name, '+') ? "" : "+";
+
+    /* Check if the target is dom0, which requires special handling. */
+    bool target_is_dom0 = strcmp(target, "@adminvm") == 0 ||
+                          strcmp(target, "dom0") == 0;
```

Comment:
The crash is because `target_domain` is NULL at this point. Which is expected for denied call. It should have been aborted earlier, `parse_policy_response()` lost the final
```
if (result != 0)
   _exit(126);
```
In fact, it seems to completely ignore actual allow/deny response, as long as it has correct syntax...